modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+
+==== Error: Variable is Referenced More Than Once
+
+Validation fails if a script attempts to reference the same variable more than once.
+
+The following script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section. Validation fails because the script attempts to reference the `payload` variable more than once.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json
+output application/json
+---
+ {
+     family: payload.family filter (member) -> member.age > 3,
+     name: payload.name
+ }
+----
+
+The script fails with the following error:
+
+[source,error,linenums]
+----
+4| input payload application/json streaming=true
+         ^^^^^^^
+Parameter `payload` is not stream capable.
+Reasons:
+ - Variable payload is referenced more than once. Locations:
+---------------------------
+
+8|      family: payload.family filter (member) -> member.age > 3,
+                ^^^^^^^
+---------------------------
+
+9|      name: payload.name
+              ^^^^^^^ at
+4| input payload application/json streaming=true
+----
+
+
+==== Error: Wrong Scope Reference
+
+Validation fails if a script attempts to reference a variable from a scope that is different from the scope in which the variable is defined.
+
+The following script fails because the `payload` variable is referenced from within the lambda expression `[1,2,3] map ((item, index) -> payload)`. Even if the expression is `[1] map ((item, index) -> payload`, streaming fails because `payload` is in the wrong scope.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json
+output application/json
+---
+[1,2,3] map ((item, index) -> payload)
+---
+
+The example fails with the following error:
+
+[source,error,linenums]
+----
+4| input payload application/json
+         ^^^^^^^
+Parameter `payload` is not stream capable.
+Reasons:
+ - Variable payload is referenced in a different scope from where it was defined. Locations:
+---------------------------
+
+9| [1,2,3] map ((item, index) -> payload)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ at
+4| input payload application/json
+----
+
+[[stream_output]]
+== Streaming Output
+
+After processing streamed data, you can stream the output directly to another message processor. To facilitate this behavior, use the `deferred` writer property in the output directive of the DataWeave script, for example, `output application/json deferred=true`.
```

Comment:
So, "after processing streamed data" refers to ALL of the preceding narrative?

Reply:
"you can...": do we need to explain why or under what circumstances our audience wants or needs to do this?

Reply:
We were talking about streaming input before. That was explained when we talked about setting streaming=true on the data source. We showed how to make the DataWeave script act on that streamed input, and we talked about the access limitations.  Now we're talking about how to stream the output to the next component in the flow.

Reply:
When we talked about end-to-end streaming in the beginning of the doc, we said that it processed the data more quickly without the memory requirements. We explained that the deferred mode avoids saving the output to disk before passing it to the next component.